run-make-check.sh

```diff
@@ -101,6 +101,7 @@ function run() {
 
     if test -f ./install-deps.sh ; then
 	    export WITH_SEASTAR=1
+	    export FOR_MAKE_CHECK=1
```

Comment:
@tchaikov It is unfortunate that if a developer doesn't use run-make-check.sh on a fresh machine, the right dependencies might not be present.  I would expect the following to work.

install-deps.sh
do_cmake.sh
cd build
make check

Reply:
agreed. it should work. as `install-deps.sh` should install all that is needed for a developer if it is called in an interactive terminal. see https://github.com/ceph/ceph/pull/26402/commits/fd43b2f0da8fe1b92ee13fe2d573b2802868b77e#diff-47a21b3706c13e08943e223c12323aa1R261

Reply:
and `for_qa_run` will enable all qa_run dependencies *and* make_check dependencies.

Reply:
I find it confusing that the interactive (developer) use case enables `for_qa_run`. Semantically speaking, that is *not* a qa run...

Reply:
> and `for_qa_run` will enable all qa_run dependencies _and_ make_check dependencies.

How does this work? On RPM systems, from reading the code changes in this PR, I would expect `for_qa_run` to enable the `qa_run` bcond but not the `make_check` bcond...

Reply:
> Semantically speaking, that is *not* a qa run...

i am open for better names.

> I would expect `for_qa_run` to enable the `qa_run` bcond but not the `make_check` bcond...

i assume that

> OK...

implies the question was answered.

Reply:
i've killed qa_run.